components/generic/ColourSwatch.vue

```diff
@@ -0,0 +1,26 @@
+<template>
+  <span
+    class="colour-palette"
+    :style="`backgroundColor: ${hexCode}`"
+    :data-qa="`colour swatch ${label}`"
+    :aria-label="`${$t('colourSwatch')} - ${label}`"
```

Comment:
As there is text visible already for the colour name, replace this with `aria-labelledby`.